### PR TITLE
Add bootstrap.sh, fix build errors on GCC 9

### PR DESCRIPTION
**Pull request by tonyhutter**
### Description
- Add bootstrap.sh to automatically download and build depedencies
- Fix some build errors on GCC 9
- Fail in configure when MPI is not detected

Use `bootstrap.sh` to download and build all UnifyCR's dependencies.  The dependencies are built in `UnifyCR/deps`, and their headers, libs, and binaries are installed to `UnifyCR/install`.   After the dependencies are built, the script prints something like this:
```
*******************************************************************************
Dependencies are all built.  You can now build Unify with:

  export PKG_CONFIG_PATH=/home/hutter/UnifyCR/install/lib/pkgconfig
  ./configure --with-leveldb=/home/hutter/UnifyCR/install --with-gotcha=/home/hutter/UnifyCR/install --with-flatcc=/home/hutter/UnifyCR/install
  make

*******************************************************************************
```

### Motivation and Context
Easily get UnifyCR built on a spack-less system.  Fix GCC 9 build errors.  This doesn't fix the build warnings though, since they're numerous and can be addressed in a separate PR.

### How Has This Been Tested?
Ran `bootstrap.sh` and built Unify on Fedora 30.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the UnifyCR code style requirements.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted.

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			git clone $i
+		fi
+	fi
+done
+
+echo "### building bmi ###"
+cd bmi
+./prepare && ./configure --enable-shared --enable-bmi-only \
+	--prefix="$INSTALL_DIR"
+make -j $(nproc) && make install
```
Comment (adammoody) [inline bootstrap.sh]: Does $nproc get set in here, or should the user set that before calling the script?

Comment (tonyhutter) [reply bootstrap.sh]: It's not a variable; we're invoking the `nproc` command here to return the number of CPUs:
```
$ nproc
12
```

Comment (adammoody) [reply bootstrap.sh]: Oh, gotcha.  Thanks.

**Comment by adammoody** [comment]
Thanks @tonyhutter

**Pull request merged**